src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -272,29 +280,45 @@ void FancyZones::ToggleEditor() noexcept
 
     MONITORINFOEX mi;
     mi.cbSize = sizeof(mi);
-    GetMonitorInfo(monitor, &mi);
+
+    m_dpiUnawareThread.submit(OnThreadExecutor::task_t{[&]{
+      if(use_cursorpos_editor_startupscreen)
+      {
+        DPIAware::GetScreenDPIForPoint(currentCursorPos, dpi_x, dpi_y);
+      }
+      else
+      {
+        DPIAware::GetScreenDPIForWindow(foregroundWindow, dpi_x, dpi_y);
+      }
+
+      GetMonitorInfo(monitor, &mi);
+    }}).wait();
+
 
     // X/Y need to start in unscaled screen coordinates to get to the proper top/left of the monitor
     // From there, we need to scale the difference between the monitor and workarea rects to get the
     // appropriate offset where the overlay should appear.
     // This covers the cases where the taskbar is not at the bottom of the screen.
-    const auto x = mi.rcMonitor.left + MulDiv(mi.rcWork.left - mi.rcMonitor.left, 96, dpi_x);
-    const auto y = mi.rcMonitor.top + MulDiv(mi.rcWork.top - mi.rcMonitor.top, 96, dpi_y);
-
+    const auto taskbar_x_offset = MulDiv(mi.rcWork.left - mi.rcMonitor.left, DPIAware::DEFAULT_DPI, dpi_x);
+    const auto taskbar_y_offset = MulDiv(mi.rcWork.top - mi.rcMonitor.top, DPIAware::DEFAULT_DPI, dpi_y);
+    const auto x = mi.rcMonitor.left + taskbar_x_offset;
+    const auto y = mi.rcMonitor.top + taskbar_y_offset;
+    const auto width = MulDiv(mi.rcWork.right - mi.rcWork.left, DPIAware::DEFAULT_DPI, dpi_x);
+    const auto height = MulDiv(mi.rcWork.bottom - mi.rcWork.top, DPIAware::DEFAULT_DPI, dpi_y);
     // Location that the editor should occupy, scaled by DPI
-    std::wstring editorLocation = 
+    const std::wstring editorLocation = 
         std::to_wstring(x) + L"_" +
         std::to_wstring(y) + L"_" +
-        std::to_wstring(MulDiv(mi.rcWork.right - mi.rcWork.left, 96, dpi_x)) + L"_" +
-        std::to_wstring(MulDiv(mi.rcWork.bottom - mi.rcWork.top, 96, dpi_y));
+        std::to_wstring(width) + L"_" +
+        std::to_wstring(height);
 
     const std::wstring params =
         iter->second->UniqueId() + L" " +
         std::to_wstring(iter->second->ActiveZoneSet()->LayoutId()) + L" " +
         std::to_wstring(reinterpret_cast<UINT_PTR>(monitor)) + L" " +
         editorLocation + L" " +
         iter->second->WorkAreaKey() + L" " +
-        std::to_wstring(static_cast<float>(dpi_x) / 96.0f);
+        std::to_wstring(static_cast<float>(dpi_x) / DPIAware::DEFAULT_DPI);

```

Comment:
Isn't this changing the result from float to a truncated integer?

Reply:
no, `dpi_x` is already casted to `float`, so the [usual arithmetic conversions](https://en.cppreference.com/w/cpp/language/operator_arithmetic#Conversions) kick in.